### PR TITLE
MIGENG-455 Kie session 'dedicated' to Complexity.drl

### PR DESCRIPTION
**Pull request by mrizzi**
https://issues.redhat.com/browse/MIGENG-455
- moved `Complexity.drl` and `GetWorkloadInventoryReports.drl` to be used in new `WorkloadInventoryComplexityKBase` Kie base (and changed `WorkloadInventoryKBase` accordingly)
- created `AgendaFocusForComplexity.drl` 
- created `BaseIntegrationTest#createAndExecuteCommandsAndGetResults` as first step for tests refactoring
- enhanced `Utils#checkLoadedRulesNumber` to manage `foo.*` value for `packages` attribute for `kbase`
- created `WorkloadInventoryComplexityTest` for testing new `WorkloadInventoryComplexityKSession0` Kie session as a `Parameterized` test to avoid code duplication in tests

### CODE REVIEW

File: src/test/java/org/jboss/xavier/analytics/test/Utils.java [hunk 1]
```diff
@@ -91,8 +92,25 @@ public static void verifyRulesFiredNames(AgendaEventListener agendaEventListener
...
     {
-        KiePackage kiePackage = kieSession.getKieBase().getKiePackage(kiePackageName);
-        Assert.assertNotNull("No rules have been loaded from '" + kiePackageName + "' package", kiePackage);
-        assertEquals("Wrong number of rules loaded", expectedLoadedRules, kiePackage.getRules().size());
+        int actualLoadedRules;
+        if (kiePackageName.endsWith(".*"))
+        {
+            final String baseKiePackageName = kiePackageName.substring(0, kiePackageName.length() - 2);
+            Collection<KiePackage> kiePackages = kieSession.getKieBase().getKiePackages();
+            Assert.assertNotNull("No rules have been loaded from '" + kiePackageName + "' package", kiePackages);
```
Comment (carlosthe19916) [inline src/test/java/org/jboss/xavier/analytics/test/Utils.java]: @mrizzi just a suggestion, I think we can use different assert messages for line 100 and 101
```suggestion
            Assert.assertNotNull("No packages have been found in kieSession", kiePackages);
```

**Review by carlosthe19916 — COMMENTED** [review]
@mrizzi I've tested your changes locally and reviewed the code; to me, this PR looks good. I'll approve this PR as soon as we resolve the suggestion I've made for the "Assert message"

File: src/test/java/org/jboss/xavier/analytics/test/Utils.java (hunk 1 above)
Comment (mrizzi) [reply src/test/java/org/jboss/xavier/analytics/test/Utils.java]: Thanks @carlosthe19916, good idea :+1:

**Review by carlosthe19916 — APPROVED** [review]

**Review by m-brophy — COMMENTED** [review]
In WorkloadInventoryComplexityTest we populate a WorkloadInventoryReportModel and then run our rules against that. Previously, in WorkloadInventoryReportTest, we populated a VMWorkloadInventoryModel and ran from scratch against that. I can see that this new kie module is supposed to run after the others so that everything the complexity rules need have already been populated. we are effectively testing only the Complexity rules here. Do we need to continue using ComplexityTest now, or is that test file now a duplication and should be removed?

**Comment by mrizzi** [comment]
@m-brophy let's say that the tests cover two different use cases:

- `ComplexityTest`, in its constructor, loads only the `Complexity.drl` to test the rules within that file (and hence it's extending `BaseTest`)
- `WorkloadInventoryComplexityTest` (that extends `BaseIntegrationTest`) loads the new `WorkloadInventoryComplexityKSession0` Kie session to test the "real" usage of rules done from `xavier-integration` application and hence loading multiple files for rules and queries to check they all work together as expected to update the complexity of a VM and also checking, with `checkUnchangedFields` method, that no unexpected field has been changed 

So i think it's worth keep having them both.

**Comment by m-brophy** [comment]
> @m-brophy let's say that the tests cover two different use cases:
> 
> * `ComplexityTest`, in its constructor, loads only the `Complexity.drl` to test the rules within that file (and hence it's extending `BaseTest`)
> * `WorkloadInventoryComplexityTest` (that extends `BaseIntegrationTest`) loads the new `WorkloadInventoryComplexityKSession0` Kie session to test the "real" usage of rules done from `xavier-integration` application and hence loading multiple files for rules and queries to check they all work together as expected to update the complexity of a VM and also checking, with `checkUnchangedFields` method, that no unexpected field has been changed
> 
> So i think it's worth keep having them both.

Great, I'm happy with that

**Review by m-brophy — APPROVED** [review]

**Pull request merged**